### PR TITLE
Allows multiple user and group matches

### PR DESCRIPTION
**Pull request by mike-carey**
*Issue #, if available:*

*Description of changes:*

This allows multiple queries to google for user and group matches.

* Usage *

```bash
ssosync -g 'name:aws-*,email=group-1@example.com,email=group-2@example.com'
```

If none are provided, it still defaults to querying everything.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/sync.go [hunk 1]
```diff
@@ -157,20 +165,30 @@ func (s *syncGSuite) SyncUsers(query string) error {
...
+		log.WithField("query", query).Debug("get google groups")
+		groups, err := s.google.GetGroups(query)
+		if err != nil {
+			return err
+		}
+		if len(groups) < 1 {
+			log.WithField("query", query).Debug("query provided no group results")
+		}
+		googleGroups = append(googleGroups, groups...)
 	}
```
Comment (steenzout) [inline internal/sync.go:192]: won't you need to dedup the `googleGroups` when `len(queries) > 1`?

File: internal/sync.go [hunk 2]
```diff
@@ -270,22 +288,31 @@ func (s *syncGSuite) SyncGroups(query string) error {
...
+	googleGroups := []*admin.Group{}
+	for _, query := range queries {
+		log.WithField("query", query).Info("get google groups")
+		groups, err := s.google.GetGroups(query)
+		if err != nil {
+			return err
+		}
+
+		if len(groups) < 1 {
+			log.WithField("query", query).Debug("query provided no group results")
```
Comment (steenzout) [inline internal/sync.go:305]:
```suggestion
			log.WithField("query", query).Debug("query provided no group results")
			continue
```

File: internal/sync.go [hunk 3]
```diff
@@ -102,10 +106,14 @@ func (s *syncGSuite) SyncUsers(query string) error {
...
-	googleUsers, err := s.google.GetUsers(query)
-	if err != nil {
-		return err
+	for _, query := range queries {
+		users, err := s.google.GetUsers(query)
+		if err != nil {
+			return err
+		}
+		googleUsers = append(googleUsers, users...)
 	}
```
Comment (steenzout) [inline internal/sync.go:118]: won't you need to dedup `googleUsers` when len(queries) > 1?

**Comment by mike-carey** [comment]
Thanks for the feedback @steenzout.  Please let me know if this implementation works for you.  I am not able to request re-review.  Is there anything else I should do to get this merged in?

**Comment by steenzout** [comment]
> Thanks for the feedback @steenzout. Please let me know if this implementation works for you. I am not able to request re-review. Is there anything else I should do to get this merged in?

LGTM.

I'm not part of the @awslabs organization so I can't merge this either. I am also interested in the feature and performed an impromptu code review that's all.

**Comment by mike-carey** [comment]
Gotcha!!  Thank you!  I really appreciate your review.

**Pull request closed without merging**